api/src/main/java/jakarta/persistence/EntityAgent.java

```diff
@@ -0,0 +1,266 @@
+ * http://www.eclipse.org/legal/epl-2.0,
+ * or the Eclipse Distribution License v. 1.0 which is available at
+ * http://www.eclipse.org/org/documents/edl-v10.php.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+ */
+
+// Contributors:
+//     Gavin King      - 4.0
+
+package jakarta.persistence;
+
+import java.util.List;
+
+/**
+ * Interface enabling more direct control over interaction with the
+ * database than what is possible with {@link EntityManager}.
+ *
+ * <p>An instance of {@code EntityAgent} must be obtained from
+ * an {@link EntityManagerFactory}, and is able to execute
+ * persistence-related operations for entity types belonging to the
+ * associated persistence unit.
+ *
+ * <p>An {@code EntityAgent} has no associated persistence context,
+ * and works only with detached entity instances. When a method of
+ * this interface is called, any necessary interaction with the
+ * database happens immediately and synchronously. In particular,
+ * {@linkplain #update update} is an explicit operation. Since
+ * there's no {@linkplain EntityManager#flush flush operation},
+ * and since the entities themselves are detached, modifications to
+ * the entities are never automatically detected and made persistent.
+ *
+ * @since 4.0
+ *
+ * @see EntityManager
+ */
+public interface EntityAgent extends EntityHandler {
+
+    /**
+     * Insert a record.
+     * <p>
+     * If the entity {@link Id} field is declared to be generated,
+     * for example, if it is annotated {@link GeneratedValue}, the
+     * id is generated and assigned to the given instance.
+     * <p>
+     * <ul>
+     * <li>The {@link jakarta.persistence.PreInsert} callback is
+     *     triggered.
+     * <li>The {@link jakarta.persistence.PostInsert} callback is
+     *     triggered if the operation is successful.
+     * </ul>
+     *
+     * @param entity a new or removed entity instance
+     *
+     * @return The identifier of the inserted entity
+     *
+     * @throws IllegalArgumentException if the given instance is
+     *         determined to not be new or removed
+     */
+    void insert(Object entity);
+
+    /**
+     * Insert every record in the given list. The records are
+     * inserted in the order in which they occur in the given list.
+     *
+     * @param entities The entities to be inserted.
+     *
+     * @see #insert(Object)
+     */
+    void insertMultiple(List<?> entities);
+
+    /**
+     * Update a record.
+     * <p>
+     * <ul>
+     * <li>The {@link jakarta.persistence.PreUpdate} callback is
+     *     triggered.
+     * <li>The {@link jakarta.persistence.PostUpdate} callback is
+     *     triggered if the operation is successful.
+     * </ul>
+     *
+     * @param entity a detached entity instance
+     *
+     * @throws IllegalArgumentException if the given instance is
+     *         determined to not be detached
+     */
+    void update(Object entity);
```

Comment:
It would be nice to have a way of knowing the entity is found for update.
```suggestion
     * @return whether a matching entity was found in the database.
     * @throws IllegalArgumentException if the given instance is
     *         determined to not be detached
     */
    boolean update(Object entity);
```